### PR TITLE
Use sharded persister for static storers

### PR DESCRIPTION
**Pull request by ssd04**
## Reasoning behind the pull request
- User persister factory for static storers creation

## Testing procedure
- Standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: storage/factory/storageServiceFactory.go [hunk 1]
```diff
@@ -263,9 +272,18 @@ func (psf *StorageServiceFactory) createAndAddBaseStorageUnits(
 	shardId := core.GetShardIDString(psf.shardCoordinator.SelfId())
 	dbPath = psf.pathManager.PathForStatic(shardId, psf.generalConfig.StatusMetricsStorage.DB.FilePath)
 	statusMetricsDbConfig.FilePath = dbPath
+
+	dbConfigHandler = NewDBConfigHandler(psf.generalConfig.StatusMetricsStorage.DB)
```
Comment (iulianpascalau) [inline storage/factory/storageServiceFactory.go:276]: this function became too large. Maybe add a TODO to split it into smaller functions?

Comment (ssd04) [reply storage/factory/storageServiceFactory.go:277]: added todo comment

**Review by BeniaminDrasovean — APPROVED** [review]

File: storage/factory/storageServiceFactory.go [hunk 2]
```diff
@@ -124,6 +124,7 @@ func checkArgs(args StorageServiceFactoryArgs) error {
 	return nil
 }
 
+// TODO: refactor this function, split it into multiple ones
```
Comment (iulianpascalau) [inline storage/factory/storageServiceFactory.go:127]: :+1:

**Review by gabi-vuls — DISMISSED** [review]
Normal allin test: master-e134e5314d -> sharded-persister-static-s-437eb02712

--- Specific errors ---

block hash does not match 6011
wrong nonce in block 2424
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  3
Nr. of all WARNS:  1402
Nr. of new ERRORS:  3
Nr. of new WARNS:  7
Nr. of PANICS:  0

/------/

--- ERRORS ---

gcp-bel-observer-2 :
Error: error calling SaveBlock, will retry   driver = *dataindexer.dataIndexer retrial in = error = { "index": "id": "statusCode": "errorType": "version_conflict_engine_exception", "reason": version conflict, document already exists (current version "causedBy": { "type": "", "reason": "" }}\n when saving transactions, block hash nonce  2
Error: error calling SaveBlock, will retry   driver = *dataindexer.dataIndexer retrial in = error = { "index": "id": "statusCode": "errorType": "version_conflict_engine_exception", "reason": version conflict, document already exists (current version "causedBy": { "type": "", "reason": "" }}\n{ "index": "id": "statusCode": "errorType": "version_conflict_engine_exception", "reason": version conflict, document already exists (current version "causedBy": { "type": "", "reason": "" }}\n when saving transactions, block hash nonce  1

/------/

--- WARNINGS ---

gcp-bel-validator-2 :
Warn: trie recreate error:           error = getNodeFromDB error: key not found for key root =  1

gcp-bel-validator-21 :
Warn: shardProcessor.createMiniBlocks     error = shard is stuck shard =  5

gcp-bel-validator-1 :
Warn: trie recreate error:           error = getNodeFromDB error: key not found for key root =  1

/------/

File: go.mod [hunk 3]
```diff
@@ -19,7 +19,7 @@ require (
 	github.com/multiversx/mx-chain-es-indexer-go v1.4.17
 	github.com/multiversx/mx-chain-logger-go v1.0.13
 	github.com/multiversx/mx-chain-scenario-go v1.2.1
-	github.com/multiversx/mx-chain-storage-go v1.0.13
+	github.com/multiversx/mx-chain-storage-go v1.0.14-0.20231205101200-0943b1e3e634
```
[the remark this answers is not in the document]
Comment (ssd04) [reply go.mod]: added proper release

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**